pkg/querier/queryrange/results_cache.go

```diff
@@ -235,6 +307,13 @@ func (s resultsCache) get(ctx context.Context, key string) ([]Extent, bool) {
 		return nil, false
 	}
 
+	// Refreshes the cache if it contains an old proto schema.
+	for _, e := range resp.Extents {
```

Comment:
This is not explicitly related to this pr, but I'm curious why a single cache hit maps to multiple entries. Since we already split by day, why isn't there a 1:1 mapping between cache keys and Extents?

Reply:
Its because the cache key is per day for a query. However for a given query you might not ask a full day, e.g 2 days and 12 hours.

So we still store partial result as extent (12hours), if another query for the same day arrives again but still not a full day just 5h hour more, then we keep adding to the extent list. (12h+5h extent).

So basically the cache is building up by day as more request comes in, and we store result for a given day inside the extent list. Of course you could have a single extent with one full day of data when you query the first time for 30days, that's guaranteed.

This is not from me, and personally I found this very smart.

Reply:
Clever, thanks for the explanation.